### DISCUSSION TITLE
Diskless booting problem on different hardware

### DISCUSSION THREAD

**Discussion by Askyyy** [ipxe/ipxe#985] [General]
Hi,

 I managed to install windows diskless boot on a single machine with success. The problem arises when i try to boot the same image on a different machine.

On the working machine when windows starts to boot, after few seconds the ISCSI drive/target disconnects, reconnects again after 7-10 seconds and everything boots normally.
When i try to boot with some other machine (older or newer hardware doesn't matter) when windows starts to boot, after few seconds the ISCSI drive/target disconnects but doesn't reconnect again and gives BOOT_DEVICE BSOD

i don't think the problem is network drivers related because if it boots in recovery and i run wpeutil initializenetwork, it connects again to the iscsi target.

It's like windows doesn't initialize the network if i change the machine
Also if i change the boot file to ipxe-snponly-x86-64.efi (latest version from github) it never reconnects the ISCSI drive even if i use wpeutil.

Any help will be grateful
I use ISCSI Console with Tiny PXE server

**Comment by NiKiZe** [comment]
Please check the several other discussions on Windows iSCSI boot.

The driver needs to be installed, and be configured boot critical. And "Windows is special", with only a slot change (PCI location) and the driver needs to be installed/configured again.

Using the same image on multiple machines is likely only going to work if they are essentially identical.

Latest binaries is found at boot.ipxe.org.

**Reply by mcb30** [reply]
It's probably also worth mentioning https://ipxe.org/sanbootconf which addresses some of these issues.  I don't think it ever got as far as being able to cope with a PCI slot change, but I do remember doing some experiments along those lines.
